extensions/script-libraries/mime/mime.livecodescript

```diff
@@ -0,0 +1,1271 @@
+﻿script "com.livecode.script-library.mime"
+/*
+Copyright (C) 2016 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>. 
+*/
+
+on revLoadLibrary
+   if the target is not me then
+      pass revLoadLibrary
+   end if
+   
+   insert the script of me into back
+   
+   if the environment contains "development" then
+      set the _ideoverride of me to true
+      revSBAddAvailableLibrary the short name of me, "MIME"
+   end if
+   
+   __Initialize
+end revLoadLibrary
+
+on revUnloadLibrary
+   if the target is not me then
+      pass revUnloadLibrary
+   end if
+   
+   remove the script of me from back
+end revUnloadLibrary
+
+/**
+
+This script library implements portions of Multipurpose Internet Mail Extensions (MIME)
+as described in RFCs 2045 - 2049
+
+*/
+
+local sMIMETypes
+
+/**
+
+The MIME version this implemented in the library
+
+Example:
+put "MIME-Version:" && MIMEVersion() & crlf after tHeaders
+
+Description:
+Use the <MIMEVersion> function create a `MIME-Version` header.
+
+*/
+
+function MIMEVersion
+   return 1.0
+end MIMEVersion
+
+/**
+
+Determine the MIME type for a file extension
+
+Example:
+put "Content-Type:" && MIMETypeForExtension("xls") & crlf after tHeaders
+
+Parameters:
+pExtension (string):
+A file extension
+
+Description:
+The mime types listed are the default mime types in an Apache web server installation
+for a given file extension. If the file extension is not found then `application/octet-stream`
+will be returned.
+
+Returns:
+A mime type for the given file extension.
+
+*/
+
+function MIMETypeForExtension pExtension
+   local tMIMEType
+   put sMIMETypes[pExtension] into tMIMEType
+   if tMIMEType is empty then
+      return "application/octet-stream" for value
+   else
+      return tMIMEType for value
+   end if
+end MIMETypeForExtension
```

Comment:
Is it feasible to get this to first do a case-sensitive lookup of the extension, and then a case-insensitive one if the first failed?  Some people have, in the past, thought it would be a great idea to use `.C` for C++ source files and `.c` for C source files... 😖 

Please name this function with a verb.